File: examples/node/publish.mjs
// Event publishing example (Node.js SDK).
// Run: RELAYHUB_API_KEY=... node publish.mjs
//
// Requires the SDK to be built first: (cd ../../sdks/node && npm install && npm run build)
import { RelayHubClient } from "../../sdks/node/dist/index.js";

const apiKey = process.env.RELAYHUB_API_KEY;
if (!apiKey) {
  console.error("Set RELAYHUB_API_KEY (an API key with the events:write scope)");
  process.exit(1);
}

const client = new RelayHubClient({
  apiKey,
  baseUrl: process.env.RELAYHUB_BASE_URL,
});

const event = await client.events.publish(
  {
    event: "payment.success",
    environment: "test",
    payload: { order_id: "ord_123", amount: 4200 },
  },
  { idempotencyKey: "ord_123-payment-success" } // safe to retry this call with the same key
);

console.log(`Published ${event.event} -- event ${event.id}`);
console.log(`${event.delivery_jobs.length} delivery job(s) queued:`);
for (const job of event.delivery_jobs) {
  console.log(`  - ${job.id} -> endpoint ${job.endpoint_id} (${job.status})`);
}

File: examples/python/publish.py
"""
Event publishing example (Python SDK).
Run: RELAYHUB_API_KEY=... python publish.py

Requires the SDK to be installed first: pip install -e ../../sdks/python
"""

import os
import sys

from relayhub import RelayHubClient, RequestOptions

api_key = os.environ.get("RELAYHUB_API_KEY")
if not api_key:
    print("Set RELAYHUB_API_KEY (an API key with the events:write scope)", file=sys.stderr)
    sys.exit(1)

with RelayHubClient(api_key=api_key, base_url=os.environ.get("RELAYHUB_BASE_URL", "https://api.relayhub.dev/v1")) as client:
    event = client.events.publish(
        event="payment.success",
        environment="test",
        payload={"order_id": "ord_123", "amount": 4200},
        options=RequestOptions(idempotency_key="ord_123-payment-success"),
    )

    print(f"Published {event['event']} -- event {event['id']}")
    print(f"{len(event['delivery_jobs'])} delivery job(s) queued:")
    for job in event["delivery_jobs"]:
        print(f"  - {job['id']} -> endpoint {job['endpoint_id']} ({job['status']})")
